### PR TITLE
feat(dop): Project pipeline binding on merge and push events

### PR DESCRIPTION
**Pull request by kakj-go**
#### What this PR does / why we need it:
Project pipeline binding on merge and push events

#### Which issue(s) this PR fixes:

- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJpdGVyYXRpb25JRHMiOlsxMTQ2LDExNjQsMTEzMSwxMTE1LDEwOTBdLCJhc3NpZ25lZSI6WyIxMDAwNTYwIl19&id=293877&iterationID=1146&pId=292482&type=TASK)

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |       The project level pipeline can display the records generated by on merge and push events      |
| 🇨🇳 中文    |      项目级流水线可以展示 on merge 和 push 事件产生的记录        |

### CODE REVIEW

**Comment by kakj-go** [comment]
test:
```yaml
version: "1.1"
"on":
  push:
    branches:
      - master
  merge:
    branches:
      - master
stages:
  - stage:
      - custom-script:
          alias: custom-script
          description: 运行自定义命令
          version: "2.0"
          image: registry.erda.cloud/erda-actions/custom-script-action:2.0-20211224-c6b7a3e
          commands:
            - sleep 120s
```

image

image

image

File: apistructs/project_pipeline.go [hunk 1]
```diff
@@ -14,14 +14,29 @@
...
 const (
-	CategoryBuildDeploy   = "build-deploy"
-	CategoryBuildArtifact = "build-artifact"
-	CategoryOthers        = "others"
+	CategoryBuildDeploy                       = "build-deploy"
+	CategoryBuildArtifact                     = "build-artifact"
+	CategoryOthers                            = "others"
+	CICDPipelineType      ProjectPipelineType = "cicd"
+
+	SourceType = "erda"
```
Comment (Effet) [inline apistructs/project_pipeline.go:29]: separate a single code block, should not put together with PipelineCategory

Comment (Effet) [reply apistructs/project_pipeline.go:29]: and `SourceType` -> `SourceTypeErda`

Comment (Effet) [reply apistructs/project_pipeline.go:29]: and `CICDPipelineType` -> `PipelineTypeCICD`

**Review by Effet — CHANGES_REQUESTED** [review]

File: apistructs/project_pipeline.go (hunk 1 above)
Comment (kakj-go) [reply apistructs/project_pipeline.go]: done

**Review by Effet — APPROVED** [review]

**Pull request merged**